DSharpPlus.Commands/Processors/TextCommands/TextExtensions.cs

```diff
@@ -0,0 +1,136 @@
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_MentionedRoles")] private static extern void _messageCreateEventArgsMentionedRolesSetter(MessageCreatedEventArgs messageCreateEventArgs, IReadOnlyList<DiscordRole> mentionedRoles);
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_MentionedChannels")] private static extern void _messageCreateEventArgsMentionedChannelsSetter(MessageCreatedEventArgs messageCreateEventArgs, IReadOnlyList<DiscordChannel> mentionedChannels);
+
+    [UnsafeAccessor(UnsafeAccessorKind.Constructor)] private static extern DiscordMessage _messageConstructor();
+    [UnsafeAccessor(UnsafeAccessorKind.Constructor)] private static extern DiscordMessage _messageConstructor(DiscordMessage message);
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_Content")] private static extern void _messageContentSetter(DiscordMessage message, string content);
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_Channel")] private static extern void _messageChannelSetter(DiscordMessage message, DiscordChannel channel);
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_ChannelId")] private static extern void _messageChannelIdSetter(DiscordMessage message, ulong channelId);
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_guildId")] private static extern void _messageGuildIdSetter(DiscordMessage message, ulong? guildId);
+    [UnsafeAccessor(UnsafeAccessorKind.Method, Name = "set_Author")] private static extern void _messageAuthorSetter(DiscordMessage message, DiscordUser author);
+    [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "mentionedUsers")] private static extern ref List<DiscordUser> _messageMentionedUsersSetter(DiscordMessage message);
+    [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "mentionedRoles")] private static extern ref List<DiscordRole> _messageMentionedRolesSetter(DiscordMessage message);
+    [UnsafeAccessor(UnsafeAccessorKind.Field, Name = "mentionedChannels")] private static extern ref List<DiscordChannel> _messageMentionedChannelsSetter(DiscordMessage message);
+
+    /// <summary>
+    /// Creates a shallow copy of a Discord message.
+    /// </summary>
+    /// <param name="message">The message to copy.</param>
+    /// <returns>A shallow copy of the message.</returns>
+    public static DiscordMessage Copy(this DiscordMessage message) => _messageConstructor(message);
+
+    /// <summary>
+    /// Creates a fake message created event args object with the specified content.
+    /// </summary>
+    /// <param name="context">The command context to use for the message.</param>
+    /// <param name="message">The message to use as a base for the new message.</param>
+    /// <param name="content">The content of the new message.</param>
+    /// <returns>The created message created event args object.</returns>
+    public static MessageCreatedEventArgs CreateFakeMessageEventArgs(CommandContext context, DiscordMessage message, string content)

```

Comment:
Also is it out of the question to make this method asynchronous, as to avoid the horror that is `GetAwaiter().GetResult()`? Generally cache ~should~ be populated by the message (especially in the case of fetching), but this may not always be the case, and should the client have to go and grab the user for any reason... ![:explosion:](https://cdn.discordapp.com/emojis/1105773806531919943.webp?size=24&quality=lossless)

Reply:
Honestly I forgot that I had `GetAwaiter().GetResult()` (though "horror" is a bit... much imo). Regardless I think we should make it async.